onadata/apps/api/viewsets/open_data_viewset.py

```diff
@@ -53,12 +53,30 @@ def process_tableau_data(data, xform):
     with the column header fields for the same form.
     Handles Flattenning repeat data for tableau
     """
+
+    def get_xpath(key, nested_key):
+        val = nested_key.split('/')
+        nested_key_diff = val[len(key.split('/')):]
+        xpaths = key + f'[{index}]/' + '/'.join(nested_key_diff)
+        return xpaths
+
+    def unpack_select_multiple_data(key, value, data):
```

Comment:
Not really sure about the function name here since as mentioned in my previous comment, the main functionality, and correct me if I'm wrong, is to update the `data` variable using choices values. How about we rename it to `get_updated_data` and make ensure that `data` is returned?
```python
...
if isinstance(value, str) and data:
    choices = value.split(" ")
    for choice in choices:
        xpaths = f'{key}/{choice}'
        data[xpaths] = choice

return data
```
Then we can then assign the `data` variable with result from `get_updated_data` like the following:

```python
...
if qstn_type == MULTIPLE_SELECT_TYPE:
    data = get_updated_data(xpaths, nested_val, data)
```